cmd/oras/internal/option/remote.go

```diff
@@ -273,6 +281,12 @@ func (opts *Remote) NewRepository(reference string, common Common) (repo *remote
 	if repo.Client, err = opts.authClient(hostname, common.Debug); err != nil {
 		return nil, err
 	}
+	logger = logger.WithField("hostname", hostname)
```

Comment:
It is better to name the field as `registry` instead of `hostname` since `hostname := repo.Reference.Registry`.

It happens to `docker.io`. The registry name is `docker.io` but the hostname is `registry-1.docker.io`.